aten/src/ATen/native/vulkan/api/DynamicShaderInfo.cpp

```diff
@@ -0,0 +1,88 @@
+#include <ATen/native/vulkan/api/DynamicShaderInfo.h>
+#include <c10/util/Exception.h>
+#include <c10/util/irange.h>
+#include <fmt/format.h>
+
+#include <unistd.h>
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+
+namespace at::native::vulkan::api {
+
+namespace {
+int count_layouts(std::string_view src) {
+  int count = 0;
+  if (src.empty()) {
+    return count;
+  }
+  auto pos = src.find("layout(set");
+  while (pos != std::string::npos) {
+    if (pos == 0 || src[pos - 1] == '\n') {
+      count++;
+    }
+    pos = src.find("layout(set", pos + 1);
+  }
+  return count;
+}
+} // namespace
+
+DynamicShaderInfo compile_glsl(std::string name, std::string_view src) {
```

Comment:
Btw, there is a programmatic interface for GLSL -> SPIR-V compilation. [We actually used it at one point](https://github.com/pytorch/pytorch/blob/1c12cbea90db347cdbd5b82f8e80e6eafba5019b/aten/src/ATen/native/vulkan/api/Shader.cpp#L69), but I removed it because doing SPIR-V compilation ahead of time reduced cold start time and I also wanted to cut down on dependencies. But we should definitely bring it back to do the JIT shader compilation.

As a prototype shell out works fine though (y)